l2geth/core/types/receipt_test.go

```diff
@@ -281,3 +281,145 @@ func clearComputedFieldsOnLog(t *testing.T, log *Log) {
 	log.TxIndex = math.MaxUint32
 	log.Index = math.MaxUint32
 }
+
+func TestDecodeReceiptsBytes(t *testing.T) {
+
+	FeeScalar := &big.Float{}
+	tx := NewTransaction(1, common.HexToAddress("0x4932"), big.NewInt(math.MaxUint32), math.MaxUint64, big.NewInt(math.MaxUint32), []byte{0x11})
+	receipt := &Receipt{
+		PostState:         common.Hash{1}.Bytes(),
+		Status:            ReceiptStatusSuccessful,
+		CumulativeGasUsed: 1800,
+		Logs: []*Log{
+			{Address: common.BytesToAddress([]byte("0x112"))},
+			{Address: common.BytesToAddress([]byte{0x01, 0x11})},
+		},
+		TxHash:           tx.Hash(),
+		ContractAddress:  common.BytesToAddress([]byte{0x02, 0x22, 0x22}),
+		GasUsed:          10000,
+		BlockHash:        common.BytesToHash([]byte{0x03, 0x33, 0x33}),
+		BlockNumber:      big.NewInt(math.MaxUint32),
+		TransactionIndex: math.MaxUint32,
+		L1GasPrice:       big.NewInt(0),
+		L1GasUsed:        big.NewInt(int64(100)),
+		L1Fee:            big.NewInt(int64(100)),
+		FeeScalar:        FeeScalar,
+		DAGasUsed:        big.NewInt(int64(10)),
+		DAFee:            &big.Int{},
+		DAGasPrice:       big.NewInt(int64(10)),
+	}
+	receipt.Bloom = CreateBloom(Receipts{receipt})
+
+	receiptByte, err := encodeAsStoredReceiptRLPV2(receipt)
+	if err != nil {
+		t.Fatalf("Error encoding receipt: %v", err)
+		return
+	}
+	var s ReceiptForStorage
+	err = rlp.DecodeReceiptsBytes(receiptByte, &s)
```

Comment:
We need to add an additional unit test for this function to consider different error types.